pkg/client/heartbeater/heartbeater.go

```diff
@@ -61,6 +65,10 @@ func (hb *heartBeater) KeepAlive(ctx context.Context) {
 				err := hb.keepAliveRequest(ctx)
 				if err != nil {
 					hb.logger.Errorf("an error occurred on keep alive %s at %s: %v\n", hb.sessionID, t.String(), err)
+					if hb.errorHandler == nil {
```

Comment:
it might be better
```
if hb.errorHandler == nil {
    hb.errorHandler(hb.sessionID, err)
}
```
because then if needed further lines could be added latter on without a dependency on the error handler being set

Reply:
you mean `if hb.errorHandler != nil` right?